pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -336,15 +336,18 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 	if len(providerID) == 0 {
 		return reconcile.Result{}, nil
 	}
-	// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+	// Ex: aws:///us-east-1e/i-078285fdadccb2eaa
+	// We always want the last entry which is the instanceID,
+	// and the first which is the provider name.
 	providerTokens := strings.Split(providerID, "/")
 	instanceID := providerTokens[len(providerTokens)-1]
 	if len(instanceID) == 0 {
 		return reconcile.Result{}, nil
 	}
+	providerName := strings.TrimSuffix(providerTokens[0], ":")
```

Comment:
I think it'd be better if we get the `provider` once when WMCO starts. Also machine object may not be ideal to get the provider info from. What do you think of [this](https://github.com/ravisantoshgudimetla/windows-machine-config-operator-1/commit/a54bb964e01a4d8e3ad322548ca4f5083ed21147)?

Reply:
yeah I think that's cleaner, but it's quite the large refactor. Would you be ok if we do that as a follow-up to this PR?

Reply:
@ravisantoshgudimetla requests like this should be done earlier in the review cycle. Lets get this merged and do this in a follow PR otherwise this will block the community operator release.